### PR TITLE
add support for QR Code for pairing 

### PR DESCRIPTION
**Pull request by yfre**
add support for QRCode pairing. 
it consists of 2 parts:
- add setup code to mDNS advertisement 
- generate SetupURI that is the basis for QR code. 

fix for #121

### CODE REVIEW

**Comment by yfre** [comment]
@ccutrer  please review

File: src/main/java/io/github/hapjava/server/impl/crypto/HAPSetupCodeUtils.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    StringBuilder builder = new StringBuilder();
+    while (count-- != 0) {
+      int character = (int) (Math.random() * ALPHA_NUMERIC_STRING.length());
+      builder.append(ALPHA_NUMERIC_STRING.charAt(character));
+    }
+    return builder.toString();
+  }
+
+  public static String generateSetupId() {
+    String setupId =
```
Comment (ccutrer) [inline src/main/java/io/github/hapjava/server/impl/crypto/HAPSetupCodeUtils.java:20]: why aren't you just calling randomAlphaNumeric here?

Comment (yfre) [reply src/main/java/io/github/hapjava/server/impl/crypto/HAPSetupCodeUtils.java:20]: right, i have copy randomAlphaNumeric from somewhere first and then it was looking for me to monstrous for just 4 random characters. 
but you are right, i should either use it and remove it.

Comment (yfre) [reply src/main/java/io/github/hapjava/server/impl/crypto/HAPSetupCodeUtils.java]: done

**Pull request merged**